scarlet/initialization.py

```diff
@@ -169,29 +169,19 @@ def trim_morphology(center_index, morph, bg_thresh):
     bbox = Box.from_bounds((bottom, top), (left, right))
     return morph, bbox
 
-
-def init_extended_source(
-    sky_coord,
+def morph_init(sky_coord,
     frame,
     observations,
+    seds,
```

Comment:
This should be an optional parameter because it only is necessary when `coadd is None`. If so, you can check with an `assert` that `seds is not None`